core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -108,30 +118,109 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
   def getOrElseF[B >: A](default: => F[B])(implicit F: Monad[F]): F[B] =
     F.flatMap(value)(_.fold(default)(F.pure))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT(List(Some(2), None, Some(414), None, None))
+   *  scala> optionT.collect{ case i: Int => i == 2 }
+   *  res0: OptionT[List, Boolean] = OptionT(List(Some(true), None, Some(false), None, None))
+   * }}}
+   */
   def collect[B](f: PartialFunction[A, B])(implicit F: Functor[F]): OptionT[F, B] =
     OptionT(F.map(value)(_.collect(f)))
 
   def exists(f: A => Boolean)(implicit F: Functor[F]): F[Boolean] =
     F.map(value)(_.exists(f))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT(List(Some(2), None, Some(414), None, None))
+   *  scala> optionT.filter(el => (el % 2 == 0))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(414), None, None))
+   *
+   *  scala> optionT.filter(el => (el % 3 == 0))
+   *  res1: OptionT[List, Int] = OptionT(List(None, None, Some(414), None, None))
+   * }}}
+   */
   def filter(p: A => Boolean)(implicit F: Functor[F]): OptionT[F, A] =
     OptionT(F.map(value)(_.filter(p)))
 
+  /**
+   * Example:
```

Comment:
I think we should mention this is for `for` comprehensions which desugar using withFilter. Otherwise it is identical to filter.

Reply:
I've added a brief description.